File: trunk/quests/qvic/Aganetti_the_Keeper.pl
sub EVENT_SAY {
  if ($text=~/hail/i) {
    quest::emote("gestures to the end of the cavernous hallway.");
    quest::say("Beyond that corner lies the entrance to the forgotten chapel of Inktu'ta. The force of the blast from the portal's implosion tore a hole in the mountain, revealing the way to this place. Some things should remain unknown. Inktu'ta should have stayed buried forever. Believe me $name only a fool would disturb the halls of this cursed chapel. Will you [heed my warning] or do you wish to [continue]...?");
  } elsif ($text=~/continue/i) {
    #if cynosure is up do not proceed
    if(!$entity_list->GetMobByNpcTypeID(295140) && !$entity_list->GetMobByNpcTypeID(295149)) {
      quest::say("Very well! If you will not heed my warning, then you have chosen to suffer the consequences. I grant you entrance to Inktu'ta.");
      #allow a GM to request the instance and if in a raid bring the entire raid with him
      #$raid = $client->GetRaid();
      #if($raid) {
      if($status > 79) { 
        if(!defined($qglobals{inktutalockout})) {
          $InInstance = quest::GetInstanceID("inktuta",0);
          if($InInstance == 0){
            $NewInstance = quest::CreateInstance("inktuta", 0, 21600);
            if($raid) {
              quest::AssignRaidToInstance($NewInstance);
              #create status global
              quest::targlobal($NewInstance.'_inktuta',0,"H6",296070,0,296);
              quest::say("Instance added.");
            } elsif($status >79) {
              quest::AssignToInstance($NewInstance);
              #create status global
              quest::targlobal($NewInstance.'_inktuta',0,"H6",296070,0,296);
              quest::say("Instance added.");
            }
          } else {
            $client->Message(13, "You are already in an instance!");
          }
        } else {
          $client->Message(13,"You have recently completed a raid.");
        }
      } else {
        $client->Message(13, "You are not in a raid!");
      }
    } else {
      #need live text for when cynosure is up.
      quest::say("The power of the Muramites is blocking the entrance to the temple.");
    }
  }
}

File: trunk/quests/qvic/Aganetti_the_Keeper.lua
-- Aganetti_the_Keeper (295065)
-- Request NPC for Inktu`Ta

function event_say(e)
	if(e.message:findi("hail")) then
		e.self:Emote("gestures to the end of the cavernous hallway.");
		e.self:Say("Beyond that corner lies the entrance to the forgotten chapel of Inktu`Ta. The force of the blast from the portal's implosion tore a hole in the mountain, revealing the way to this place. Some things should remain unknown. Inktu'ta should have stayed buried forever. Believe me " .. e.other:GetName() .. " only a fool would disturb the halls of this cursed chapel. Will you [heed my warning] or do you wish to [" .. eq.say_link("continue",false,"continue") .. "]?");
	elseif(e.message:findi("continue")) then
		--if Cynosure is up do not proceed
		if(eq.get_entity_list():IsMobSpawnedByNpcTypeID(295140) == false and eq.get_entity_list():IsMobSpawnedByNpcTypeID(295149) == false) then
			local inktuta_globals = {
				{ "Ink_Kelekdrix", "Kelekdrix, Herald of Trushar" },
				{ "Ink_Mites", "Stonemite Event" },
				{ "Ink_Golems", "Golems" },
				{ "Ink_Callers", "Cursecaller Event" },
				{ "Ink_Noq", "Noqufiel" }
			} 
			local instance_requests = require("instance_requests");
			local request = instance_requests.ValidateInstanceRequest("inktuta", 54, e.other, inktuta_globals);
			if (request.valid) then
				local instance_id = eq.create_instance("inktuta", 0, 21600);
				eq.set_global(instance_id.."_inktuta_bit",tostring(request.flags),7,"H6");
				eq.assign_raid_to_instance(instance_id);
				e.self:Say("Very well! If you will not heed my warning, then you have chosen to suffer the consequences. I grant you entrance to Inktu`Ta.");
			end
		else
			--need live text for when cynosure is up
			e.other:Message(13, "The power of the Muramites is blocking the entrance to the temple.");
		end
	end
end